Classes/Resource/Target/ProtectedResourceTarget.php

```diff
@@ -180,14 +180,14 @@ protected function shouldIncludeSecurityContext()
     protected function detectResourcesBaseUri()
     {
         $request = $this->getHttpRequest();
-        if (!$request instanceof HttpRequest) {
+        if (!$request instanceof HttpRequestInterface) {
             return $this->defaultBaseUri;
         }
-        return (string)$request->getBaseUri();
+        return (string)$request->getUri();
```

Comment:
This won't return the base Uri but the current one. We might want to use the `BaseUriProvider` here

Reply:
Thank you for your feedback. I fixed this to use the `RequestInformationHelper::generateBaseUri` to get the base uri.